spec/models/manageiq/providers/autosde/storage_service_spec.rb

```diff
@@ -0,0 +1,103 @@
+describe ManageIQ::Providers::Autosde::StorageManager::StorageService do
+  let(:ems) do
+    FactoryBot.create(
+      :autosde_storage_manager,
+      :with_autosde_credentials,
+      :hostname     => Rails.application.secrets.autosde[:appliance_host],
+      :capabilities => {
+        :cap1 => [{'uuid' => "111", 'value' => "True"}, {'uuid' => "222", 'value' => "False"}],
+        :cap2 => [{'uuid' => "333", 'value' => "True"}, {'uuid' => "444", 'value' => "False"}],
+      }
+    )
+  end
+
+  let(:test_resource1) do
+    FactoryBot.create(:storage_resource, :ext_management_system => ems, :ems_ref => "555")
+  end
+
+  let(:test_resource2) do
+    FactoryBot.create(:storage_resource, :ext_management_system => ems, :ems_ref => "666")
+  end
+
+  let(:test_resource3) do
+    FactoryBot.create(:storage_resource, :ext_management_system => ems, :ems_ref => "777")
+  end
+
+  before(:each) do
+    @test_service = FactoryBot.create(
+      :storage_service,
+      :type                  => "ManageIQ::Providers::Autosde::StorageManager::StorageService",
+      :ext_management_system => ems,
+      :storage_resources     => [test_resource1, test_resource2],
+      :capabilities          => {'cap1' => "True", 'cap2' => "False"}
+    )
+  end
```

Comment:
Minor but it is preferable to use `let(:test_service)` (or `let!` if you need the object to be created prior to being referenced)
```suggestion
  let!(:test_service) do
    FactoryBot.create(
      :storage_service,
      :type                  => "ManageIQ::Providers::Autosde::StorageManager::StorageService",
      :ext_management_system => ems,
      :storage_resources     => [test_resource1, test_resource2],
      :capabilities          => {'cap1' => "True", 'cap2' => "False"}
    )
  end
```